server/sonar-server/src/main/java/org/sonar/server/component/ComponentService.java

```diff
@@ -58,64 +47,20 @@
 @ComputeEngineSide
 public class ComponentService {
   private final DbClient dbClient;
-  private final I18n i18n;
   private final UserSession userSession;
-  private final System2 system2;
-  private final ComponentFinder componentFinder;
   private final ProjectMeasuresIndexer projectMeasuresIndexer;
   private final ComponentIndexer componentIndexer;
 
-  public ComponentService(DbClient dbClient, I18n i18n, UserSession userSession, System2 system2, ComponentFinder componentFinder, ProjectMeasuresIndexer projectMeasuresIndexer,
+  public ComponentService(DbClient dbClient, UserSession userSession, ProjectMeasuresIndexer projectMeasuresIndexer,
     ComponentIndexer componentIndexer) {
     this.dbClient = dbClient;
-    this.i18n = i18n;
     this.userSession = userSession;
-    this.system2 = system2;
-    this.componentFinder = componentFinder;
     this.projectMeasuresIndexer = projectMeasuresIndexer;
     this.componentIndexer = componentIndexer;
   }
 
-  public ComponentDto getByKey(String key) {
-    DbSession session = dbClient.openSession(false);
-    try {
-      return getByKey(session, key);
-    } finally {
-      session.close();
-    }
-  }
-
-  @CheckForNull
-  public ComponentDto getNullableByKey(String key) {
-    DbSession session = dbClient.openSession(false);
-    try {
-      Optional<ComponentDto> component = dbClient.componentDao().selectByKey(session, key);
-      return component.orNull();
-    } finally {
-      session.close();
-    }
-  }
-
-  public ComponentDto getNonNullByUuid(String uuid) {
-    DbSession session = dbClient.openSession(false);
-    try {
-      return dbClient.componentDao().selectOrFailByUuid(session, uuid);
-    } finally {
-      session.close();
-    }
-  }
-
-  public Optional<ComponentDto> getByUuid(String uuid) {
-    DbSession session = dbClient.openSession(false);
-    try {
-      return dbClient.componentDao().selectByUuid(session, uuid);
-    } finally {
-      session.close();
-    }
-  }
-
-  public void updateKey(DbSession dbSession, String projectOrModuleKey, String newKey) {
-    ComponentDto component = componentFinder.getByKey(dbSession, projectOrModuleKey);
+  // TODO should be moved to ComponentUpdater
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)